wwwapp/models.py

```diff
@@ -22,6 +21,27 @@ class UserProfile(models.Model):
     def is_participating_in(self, year):
         return self.status_for(year) == 'Z' \
                or Workshop.objects.filter(type__year=year, lecturer=self, status='Z').exists()
+
+    def all_participation_years(self) -> list:
```

Comment:
inb4 @grzegorz225 "not enough type info, should be List[int]" (or something like that, I don't remember :P)